### DISCUSSION TITLE
More than 128 atoms on gpu

### DISCUSSION THREAD

**Discussion by JonathanSchmidt1** [deepmodeling/deepmd-kit#2705] [Q&A]
Hi everyone,

My training systems have 320-360 atoms. In the docs https://docs.deepmodeling.com/projects/deepmd/en/master/train/gpu-limitations.html it is written that the limit for gpu is 128, is this just a memory issue and I will be fine if I use a sufficient gpu or is  it actually hard-coded somewhere  and the results will be wrong with more atoms.

thank you very much, Jonathan

**Comment by owen-rett** [comment]
My understanding is that its only an issue if the number of types of atoms is greater than 128, not just the number of atoms. E.g. having 96 Carbon atoms and 192 Oxygen atoms in a cell shouldn't be an issue. I've certainly trained on systems with more than 128 individual atoms (albeit with 3 types of atoms) with no issue.

**Reply by JonathanSchmidt1** [reply]
Thank you very much for the answer. I just missunderstood it.
